### PR TITLE
Make the donation form submittable

### PR DESCRIPTION
**Pull request by gbirke**
Add ViewHandlers that dynamically adjust the functionality of the submit buttons depending on the validation status.

Add TODOs where work needs to be done in case the donation does not validate on the server side (as it is the case at the moment, because the amount is not sent properly).

### CODE REVIEW

**Comment by JeroenDeDauw** [comment]
> the functionality of the submit buttons

I had a confused there for a second :)

File: app/templates/Form1.html.twig [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<form name="donForm" id="donForm" method="POST" action="#">
+<form name="donForm2" id="donForm1" method="POST" action="#">
```
Comment (JeroenDeDauw) [inline app/templates/Form1.html.twig]: ?

File: src/Factories/FunFunFactory.php [hunk 2]
```diff
@@ -852,7 +852,11 @@ public function getDonationConfirmationPageSelector() {
 	}
 
 	public function newDonationFormViolationPresenter() {
-		return new DonationFormViolationPresenter( $this->getLayoutTemplate( 'DonationForm.html.twig' ) );
+		$template = $this->getLayoutTemplate( 'DisplayPageLayout.twig' );
+		// TODO make this dependent on the 'form' value from the HTTP POST request
+		// (we need different form pages for A/B testing)
```
Comment (JeroenDeDauw) [inline src/Factories/FunFunFactory.php]: We have stuff in place already for showing different donation confirmation pages. There no HTTP parameter is used. Does that make sense?

File: app/templates/DonationForm.html.twig [hunk 3]
```diff
@@ -142,6 +163,27 @@
                     {
                         viewHandler: WMDE.View.createSlidingVisibilitySwitcher( $( '.personal-data-full' ), /firma|person/ ),
                         stateKey: 'formContent.addressType'
+                    },
+                    {
+                        viewHandler: WMDE.View.createValueDependentEventHandler( $( '#continueFormSubmit1' ), 'click', [
```
Comment (JeroenDeDauw) [inline app/templates/DonationForm.html.twig]: What's the deal with this 1 and 2 stuff? The first and second steps on the client?

File: app/js/tests/view_handlers/test_value_dependent_event_handler.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+var test = require( 'tape' ),
+	sinon = require( 'sinon' ),
+	createValueDependentEventHandler = require( '../../lib/view_handler/value_dependent_event_handler' ).createHandler;
+
+test( 'When the handler is created it attaches it attaches an event listener', function ( t ) {
```
Comment (JeroenDeDauw) [inline app/js/tests/view_handlers/test_value_dependent_event_handler.js]: 2x it attaches

**Comment by gbirke** [comment]
The JS classes must be rewritten and one commit was cherry-picked and modified into another commit, so I'm closing this to avoid confusion.

**Pull request closed without merging**